### PR TITLE
Restructure Awesome books code with ES6

### PR DESCRIPTION
**Pull request by lenkon**
Project name: Awesome books - with ES6

Project description: In this project, the Awesome books app code will be restructured using ES6, and ES6 syntax. The goal is to make it more organized by using modules.

Project requirements: 
- Setup project:
  - Create a new directory and initialize it with git.
  - Use `npm init -y` command to create package.json file.
  - Create the entry point for the JavaScript code called `index.js`
  - Create `index.css` file for styles.
  - Create the main `index.html` file. Link the `index.js` and `index.css` files there.
  - Create a directory called `modules`.
- Re-create the _Awesome books_ app following the rules:
  - Use ES6 syntax.
  - Divide the code into modules. Save each module in a separate file in the `modules` dir. Import modules in the `index.js` file. For this exercise - the more modules the better!
  - Keep all functionalities of the app without errors.
  - Refactor the methods and functions to arrow functions.
  - Use let and const in a correct way.
 
- Display the current date below the navigation bar. 
  - For this requirement you should use a library [Luxon](https://moment.github.io/luxon/#/install?id=es6)
  - Download the script and save it to your project. 
  - Import it as a module.
  - See the article [Get the current date and time](https://moment.github.io/luxon/#/tour?id=get-the-current-date-and-time). Your task here is to use a new tool effectively with little externally curated support. Exactly as you will do in your future job.

Project requirements link: https://github.com/microverseinc/curriculum-javascript/blob/main/books/books_with_es6.md

### CODE REVIEW

**Review by leonardodiasb — CHANGES_REQUESTED** [review]
# Required Changes ♻️ ⚠️ 🚦 

Hi @lenkon, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

mario_ladders

## Good points 👍 

 - Very organized. ✔️
 - Clean code. ✔️ 
 - Very professional PR. ✔️ 

## Aspects to improve ♻️

- [x] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by lenkon** [comment]
Hi @leonardodiasb! Firstly, I would like to thank you for reviewing my code. And about the review, I'm having some confusion. I cannot find or see your comment or reviews. Could you please let me know where I can find your review?

File: modules/library.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    if (title.value.length > 0 && author.value.length > 0) {
+      this.bookData.push({
+        title: title.value,
+        author: author.value,
+      });
+      this.storeData();
+      this.showSavedData();
+      this.formInput.reset();
+    }
+  }
```
Comment (medaminedev66) [inline modules/library.js]: - [ ] Great job refactoring your code. However, the web app should function normally. The user is not able to add books. I tried to add a book, but I could not find it on the List page, not even in the local storage. Please make sure the book is displayed on the List page and saved to the local storage.

Note: Once you resolve the issue with adding books, please double-check that deleting books works properly before submitting a new review.

**Review by medaminedev66 — COMMENTED** [review]
Hi @lenkon,
### This is a manual review
# Request Changes♻️ 

Kindly check the comment below the review.

Cheers and Happy coding!:clap::clap::clap:

File: modules/library.js (hunk 1 above)
Comment (lenkon) [reply modules/library.js]: Hi @medaminedev66, thanks for the valuable feedback. But I would like to know whether you tested it by running Live Server for VS code or without running the server. As this code uses different modules we need to run a server like Live Server for VS Code as the Chrome browser by itself has issues with CORS policy. In my system when I test it by running the Live server, I can add, delete and see the books in the list. So, could you please recheck and let me know?

**Comment by leonardodiasb** [comment]
Hello @lenkon, I'm really sorry for not making the proper inline comment on my review. Please, feel free to reach me out on Slack. I had the same issue as @medaminedev66 when I tried to add a book. The bug:
image

**Comment by lenkon** [comment]
Hi @leonardodiasb, thanks for the response. Could you please recheck it by running VS code server and see the result? It's working fine in my system but I had to run the Live Server for the VS Code.

**Comment by lenkon** [comment]
Hi @leonardodiasb and @medaminedev66, thanks for all your feedback. I have been able to figure out the problem. The issue is that I tried to push in a non-existent array, which happens when I tried to push in the array for the first time. Now, I've fixed the issue.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+- [ ] **[Add more projects]**
+- [ ] **[Add milestones and achievements]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🤝 Contributing <a name="contributing"></a>
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](../../issues/).
```
Comment (danmainah) [inline README.md:124]: - [ ] @lenkon it would have been nice if you added a link to your issue page and ensure it works.

**Review by danmainah — APPROVED** [review]
Hi @lenkon , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ No linter errors
✔️ Github flow was correctly used
✔️  Good design

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**